lib/model.js

```diff
@@ -351,12 +351,22 @@ Model.prototype.$__handleSave = function(options, callback) {
 
     const update = delta[1];
     if (this.$__schema.options.minimize) {
-      minimize(update);
-      // minimize might leave us with an empty object, which would
-      // lead to MongoDB throwing a "Update document requires atomic operators" error
-      if (Object.keys(update).length === 0) {
-        handleEmptyUpdate.call(this);
-        return;
+      for (const updateOp of Object.values(update)) {
+        if (updateOp == null) {
+          continue;
+        }
+        for (const key of Object.keys(updateOp)) {
+          if (updateOp[key] == null || typeof updateOp[key] !== 'object') {
+            continue;
+          }
+          if (!utils.isPOJO(updateOp[key])) {
+            continue;
+          }
+          minimize(updateOp[key]);
+          if (Object.keys(updateOp[key]).length === 0) {
+            updateOp[key] = null;
+          }
```

Comment:
```suggestion
          // minimize might leave us with an empty object, which would
          // lead to MongoDB throwing a "Update document requires atomic operators" error
          if (Object.keys(updateOp[key]).length === 0) {
            updateOp[key] = null;
          }
```

unless the semantics changed, it would likely be good to keep the comment from before the PR for this case

Reply:
Semantics changed. Before, we could minimize `{ $set: { sub: {} } }` down to an empty object `{}`, and MongoDB server throws an error if you try to send `updateOne()` with empty object as `update`. With this change, `{ $set: { sub: {} } }` just minimizes down to `{ $set: { sub: null } }`, which is perfectly fine.